app/src/androidTest/java/com/github/orkest/ViewModel/profile/MockProfileViewModel.kt

```diff
@@ -54,15 +54,18 @@ class MockProfileViewModel(user: String) : ProfileViewModel(user) {
         return CompletableFuture.completedFuture(isUserFollowed.value)
     }
 
-    override fun follow(): CompletableFuture<Boolean> {
-        isUserFollowed.value = true
-        return CompletableFuture.completedFuture(true)
+
+    override fun updateCurrentUserFollowings(toFollow: Boolean): CompletableFuture<Boolean> {
+        isUserFollowed.value = toFollow
+        return super.updateCurrentUserFollowings(toFollow)
     }
 
-    override fun unfollow(): CompletableFuture<Boolean> {
-        isUserFollowed.value = false
-        return CompletableFuture.completedFuture(true)
+    override fun updateUserFollowers(toFollow: Boolean): CompletableFuture<Boolean> {
+        isUserFollowed.value = toFollow
+        return super.updateCurrentUserFollowings(toFollow)
     }
 
 
+
+
 }

```

Comment:
The methods in the MockProfileViewModel class should be consistent with the original ProfileViewModel class, so it's better to implement both follow() and unfollow() methods that call the new generic methods internally, but that's like a minor issue, it's just to say ;)